src/OpenTelemetry.AutoInstrumentation.Native/cor_profiler.cpp

```diff
@@ -177,22 +191,10 @@ HRESULT STDMETHODCALLTYPE CorProfiler::Initialize(IUnknown* cor_profiler_info_un
         }
     }
 
-    // get ICorProfilerInfo10 for >= .NET Core 3.0
```

Comment:
It seems that instead of moving the location of this code to the top, it could be just changed here to be 12: code prior to this already ensures that if .NET the version is 6 or greater.

Reply:
I think after some additional changes requesting interfaces in one place is better option.